File: Beecrowd/Beginner/1435 - 1985/1534 - Array 123/C++/1534_Array123.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int arraySize;

    while (cin >> arraySize) {
        int matrix[arraySize][arraySize];

        for (int i = 0; i < arraySize; i++) {
            matrix[i][i] = 1;
            matrix[i][arraySize - i - 1] = 2;

            for (int j = 0; j < arraySize; j++) {
                if (i != j && j != arraySize - i - 1) {
                    matrix[i][j] = 3;
                }
            }
        }

        for (int i = 0; i < arraySize; i++) {
            for (int j = 0;  j < arraySize; j++) {
                cout << matrix[i][j];
            }
            
            cout << endl;
        }
    }

    return 0;
}

File: Beecrowd/1 - Beginner/1435 - 1985/1534 - Array 123/1534_Array123.py
while True:
    try:
        arraySize = int(input())
        matrix = [[0 for _ in range(arraySize)] for _ in range(arraySize)]

        for i in range(arraySize):
            matrix[i][i] = 1
            matrix[i][arraySize - i - 1] = 2

            for j in range(arraySize):
                if i != j and j != arraySize - i - 1:
                    matrix[i][j] = 3

        for row in matrix:
            print(''.join(map(str, row)))

    except EOFError:
        break
